google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/BigtableBatchingCallSettings.java

```diff
@@ -145,9 +188,92 @@ public Builder setRetrySettings(@Nonnull RetrySettings retrySettings) {
+     * FlowController} bases on isLatencyBasedThrottlingEnabled and BatchingSettings.
+     */
+    public Builder setLatencyBasedThrottling(
+        boolean isLatencyBasedThrottlingEnabled, @Nullable Long targetRpcLatency) {
+      Preconditions.checkState(
+          this.batchingSettings != null,
+          "batchingSettings must be set before enabling / disabling throttling");
+      Preconditions.checkArgument(
+          !isLatencyBasedThrottlingEnabled || targetRpcLatency != null,
+          "target RPC latency must be set if latency based throttling is enabled");
+      Preconditions.checkArgument(
+          targetRpcLatency == null || targetRpcLatency > 0,
+          "if target RPC latency is set, it must be greater than 0");
+      this.isLatencyBasedThrottlingEnabled = isLatencyBasedThrottlingEnabled;
+      this.targetRpcLatencyMs = isLatencyBasedThrottlingEnabled ? targetRpcLatency : null;
+      if (isLatencyBasedThrottlingEnabled) {
+        // Set up a flow controller with DynamicFlowControlSettings
+        Long maxThrottlingElementCount =
+            batchingSettings.getFlowControlSettings().getMaxOutstandingElementCount();
+        Long maxThrottlingRequestByteCount =
+            batchingSettings.getFlowControlSettings().getMaxOutstandingRequestBytes();
+        if (maxThrottlingElementCount == null) {
+          long maxBulkMutateElementPerBatch = 100L;
+          long defaultChannelPoolSize = 2 * Runtime.getRuntime().availableProcessors();
+          maxThrottlingElementCount =
+              Math.min(20_000L, 10L * maxBulkMutateElementPerBatch * defaultChannelPoolSize);
+        }
+        if (maxThrottlingRequestByteCount == null) {
+          maxThrottlingRequestByteCount = 100L * 1024 * 1024;
+        }
+
+        long initialElementCount = maxThrottlingElementCount / 4;
+        long minElementCount = maxThrottlingElementCount / 100;
+        if (batchingSettings.getElementCountThreshold() != null) {
+          initialElementCount =
+              Math.max(initialElementCount, batchingSettings.getElementCountThreshold());
+          minElementCount = Math.max(minElementCount, batchingSettings.getElementCountThreshold());
+        }
+        DynamicFlowControlSettings dynamicFlowControlSettings =
+            DynamicFlowControlSettings.newBuilder()
+                .setInitialOutstandingElementCount(initialElementCount)
+                .setMaxOutstandingElementCount(maxThrottlingElementCount)
+                .setMinOutstandingElementCount(minElementCount)
+                .setInitialOutstandingRequestBytes(maxThrottlingRequestByteCount)
+                .setMinOutstandingRequestBytes(maxThrottlingRequestByteCount)
+                .setMaxOutstandingRequestBytes(maxThrottlingRequestByteCount)
+                .build();
+        this.flowController = new FlowController(dynamicFlowControlSettings);
+      } else {
+        this.flowController = new FlowController(batchingSettings.getFlowControlSettings());
+      }
```

Comment:
I think it would be better to defer flowController instantiation to Stub creation. I think that flow control should be scoped to the client instance. ie if do something like
BigtableDataSettings.Builder b = BigtableDataSettings.newBuilder(). enableBatchMutationLatencyBasedThrottling(100);

c1 = BigtableDataClient.create(b.setProjectId("p1").build())
c2 = BigtableDataClient.create(b.setProjectId("p2").build())
I should end up with separate flow controllers

and it will be less surprising that
c2 = BigtableDataClient.create(b.enableBatchMutationLatencyBasedThrottling(100).build())
also has its own flow controller

Reply:
Moved FlowController creation to EnhancedBigtableStub constructor, and added some tests for EnhancedBigtableStub.create and BigtableDataClientFactory.create